mm2src/coins/rpc_command/get_estimated_fees.rs

```diff
@@ -0,0 +1,217 @@
+//! RPCs to start/stop gas fee estimator and get estimated base and priority fee per gas
+
+use crate::eth::{EthCoin, FeePerGasEstimated};
+use crate::AsyncMutex;
+use crate::{from_ctx, lp_coinfind, MmCoinEnum, NumConversError};
+use common::executor::{spawn_abortable, AbortOnDropHandle, Timer};
+use common::log::debug;
+use common::{HttpStatusCode, StatusCode};
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::prelude::*;
+use std::sync::Arc;
+
+const FEE_ESTIMATOR_NAME: &str = "eth_gas_fee_estimator_loop";
+const ETH_SUPPORTED_CHAIN_ID: u64 = 1; // only eth mainnet is suppported (Blocknative gas platform currently supports Ethereum and Polygon/Matic mainnets.)
+                                       // To support fee estimations for other chains add a FeeEstimatorContext for a new chain
```

Comment:
Maybe `FeeEstimatorContext` should be part of `EthCoin`/`EthCoinImpl` struct instead, and the `check_if_chain_id_supported` is done on coin initialization to initialize it for the supported chains. It would be good to not have the supported chains hardcoded too but to get if it's supported or not from coin config in case a new chain is supported.

Reply:
I think we could create FeeEstimatorContext in the platform coin and set references to it when tokens are created.
As to have a setting if fee estimates are supported in the coins file would this not be error-prone to fix this setting for each token? Maybe better to have a whole gas provider api config, with chain id covering all tokens

Reply:
> I think we could create FeeEstimatorContext in the platform coin and set references to it when tokens are created.

I agree.

> As to have a setting if fee estimates are supported in the coins file would this not be error-prone to fix this setting for each token? Maybe better to have a whole gas provider api config, with chain id covering all tokens

It should be for platform coin config only.

Reply:
added to TODO in this PR